nap/utils.py

```diff
@@ -1,9 +1,36 @@
 from __future__ import unicode_literals
+from collections import namedtuple
 import itertools
+from urlparse import urlparse
+
 import six
 from six.moves import urllib
 
 
+def normalize_url(url_string):
+    QueryParam = namedtuple("QueryParam", "key value")
+    url = urlparse(url_string)
+    if not url.query:
+        return url_string
+
+    query_params = []
+    for entry in url.query.split('&'):
+        key_value = entry.split("=")
+        query_params.append(QueryParam(key=key_value[0], value=key_value[1] if len(key_value) == 2 else None))
+    query_params.sort()
```

Comment:
Could this block be equivalently replaced by `sorted(parse_qsl(url.query, keep_blank_values=True))`? Parsing query params manually with `split` has bitten us in the past.

https://docs.python.org/2/library/urlparse.html#urlparse.parse_qsl